metadata-ingestion/src/datahub/ingestion/transformer/add_dataset_tags.py

```diff
@@ -33,6 +37,26 @@ def create(cls, config_dict: dict, ctx: PipelineContext) -> "AddDatasetTags":
         config = AddDatasetTagsConfig.parse_obj(config_dict)
         return cls(config, ctx)
 
+    def create_tag_if_not_exist(
```

Comment:
we can do this without needing a graph instance. See https://github.com/datahub-project/datahub/blob/c381806110ae995dd2164305394ee4e1d131e033/metadata-ingestion/src/datahub/ingestion/api/source_helpers.py#L147 for an example

Reply:
I went through the code, so ideally We want to generate MCP to avoid dependency on the graph instance.

We do need to refactor [base_transformer.py#L183](https://github.com/datahub-project/datahub/blob/master/metadata-ingestion/src/datahub/ingestion/transformer/base_transformer.py#L183). Here the base transformer is expecting single Aspect to be generated for the given entity from extended transformer.

In base_transformer we need to change in the signature of [base_transformer.py#L210C57-L210C57](https://github.com/datahub-project/datahub/blob/master/metadata-ingestion/src/datahub/ingestion/transformer/base_transformer.py#L210C57-L210C57) to Union[Aspect, List[Aspect], List[MCP]) and accordingly generates the RecordEnvelope.

Please let me know your thoughts

Reply:
Instead of changing the signature of `transform_aspect`, can we add a "handle_end_of_stream" method that is a no-op in most cases, but the add tag transformer implements it and emits the necessary MCPs?